meta_tools/utils/utils_script_gen.py

```diff
@@ -0,0 +1,55 @@
+from pathlib import Path
+import py_compile
+
+
+def save_script(output_file: Path, code: str) -> bool:
+    """
+Compiles a Python script to a specified file and saves it.
+
+Parameters:
+output_file : Path
+    The path to the file where the Python script will be saved. This should
+    be a `Path` object pointing to the desired file location, including
+    the filename and extension (e.g., 'script.py').
+
+code : str
+    The Python code to be written to the file. This should be a string containing
+    valid Python code that will be saved and compiled.
+
+Returns:
+bool:
+    Returns `True` if the script was successfully written and compiled.
+    Returns `False` if an error occurred during the file writing or compilation.
+
+Raises:
+IOError: If there is an error writing the code to the file (e.g., permission denied, invalid path).
+
+py_compile.PyCompileError: If there is an error compiling the Python script (e.g., syntax error in the code).
+"""
+    if output_file.exists():
+        while True:
+            overwrite = input(
+                f"The file {output_file} already exists. Do you want to overwrite it? (y/n): ").strip().lower()
+            if overwrite in {'y', 'n'}:
```

Comment:
I don't know... Can you remind me of the use case for overwriting an existing file here? If there is a good use-case, then I have no objections but if this is just something we think might be nice-to-have then I think we should consider not giving the option. 

Let me know what you think?

Reply:
This is to let the user revert back to the original script if they break something. Not too important tbh. Can remove.